### ISSUE TITLE
Unable to download from Souncloud

### ISSUE THREAD

**Issue by jeremycherfas** [rg3/youtube-dl#12376]
## Please follow the guide below

- You will be asked some questions and requested to provide some information, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *issue* (like that [x])
- Use *Preview* tab to see how your issue will actually look like

---

### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2017.03.06*. If it's not read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [ x] I've **verified** and **I assure** that I'm running youtube-dl **2017.03.06**

### Before submitting an *issue* make sure you have:
- [x ] At least skimmed through [README](https://github.com/rg3/youtube-dl/blob/master/README.md) and **most notably** [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [ ] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones

### What is the purpose of your *issue*?
- [ x] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### The following sections concretize particular purposed issues, you can erase any section (the contents between triple ---) not applicable to your *issue*

---

### If the purpose of this *issue* is a *bug report*, *site support request* or you are not completely sure provide the full verbose output as follows:

Add `-v` flag to **your command line** you run youtube-dl with, copy the **whole** output and insert it here. It should look similar to one below (replace it with **your** log inserted between triple ```):
```
Jeremys-iMac:~ jeremycherfas$ youtube-dl --verbose https://soundcloud.com/geofft-1/chicken-the-charismatic-supermarkets-a-cultural-economy-of-power-quinoa-prof-jane-dixon
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'--verbose', u'https://soundcloud.com/geofft-1/chicken-the-charismatic-supermarkets-a-cultural-economy-of-power-quinoa-prof-jane-dixon']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2017.03.06
[debug] Python version 2.7.10 - Darwin-15.6.0-x86_64-i386-64bit
[debug] exe versions: none
[debug] Proxy map: {}
[soundcloud] geofft-1/chicken-the-charismatic-supermarkets-a-cultural-economy-of-power-quinoa-prof-jane-dixon: Resolving id
[soundcloud] geofft-1/chicken-the-charismatic-supermarkets-a-cultural-economy-of-power-quinoa-prof-jane-dixon: Downloading info JSON
ERROR: Unable to download JSON metadata: HTTP Error 401: Unauthorized (caused by HTTPError()); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 496, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 2093, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 437, in open
    response = meth(req, response)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 550, in http_response
    'http', request, response, code, msg, hdrs)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 475, in error
    return self._call_chain(*args)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 409, in _call_chain
    result = func(*args)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 558, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
...
<end of log>
```

---

**Comment by frankie-loves-jesus** [comment]
Was just about to log on to report this. Good going!

**Comment by sedmelluq** [comment]
It's a client ID change. About time to fetch it from the app.js instead? I've seen it change twice in the last few months already.

**Comment by varunest** [comment]
@sedmelluq Yeah exactly.

**Comment by varunest** [comment]
@sedmelluq Do you know what is the new client ID?

**Comment by Naereen** [comment]
Exactly the same issue, on another example of SoundClound track:

```bash
$ youtube-dl -v  --youtube-skip-dash-manifest --format worst --output "%(title)s.%(ext)s" --extract-audio --console-title --audio-format=mp3 --no-overwrites --  'https://soundcloud.com/penofchaos/naheulbeuk-hors-serie-les-blagues'
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'-v', u'--youtube-skip-dash-manifest', u'--format', u'worst', u'--output', u'%(title)s.%(ext)s', u'--extract-audio', u'--console-title', u'--audio-format=mp3', u'--no-overwrites', u'--', u'https://soundcloud.com/penofchaos/naheulbeuk-hors-serie-les-blagues']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2017.03.06
[debug] Python version 2.7.12+ - Linux-4.8.0-39-generic-x86_64-with-Ubuntu-16.10-yakkety
[debug] exe versions: avconv 3.0.7-0ubuntu0.16.10.1, avprobe 3.0.7-0ubuntu0.16.10.1, ffmpeg 3.0.7-0ubuntu0.16.10.1, ffprobe 3.0.7-0ubuntu0.16.10.1
[debug] Proxy map: {}
[soundcloud] penofchaos/naheulbeuk-hors-serie-les-blagues: Resolving id
[soundcloud] penofchaos/naheulbeuk-hors-serie-les-blagues: Downloading info JSON
ERROR: Unable to download JSON metadata: HTTP Error 401: Unauthorized (caused by HTTPError()); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
  File "~/bin/youtube-dl/youtube_dl/extractor/common.py", line 496, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "~/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 2093, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/usr/lib/python2.7/urllib2.py", line 435, in open
    response = meth(req, response)
  File "/usr/lib/python2.7/urllib2.py", line 548, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib/python2.7/urllib2.py", line 473, in error
    return self._call_chain(*args)
  File "/usr/lib/python2.7/urllib2.py", line 407, in _call_chain
    result = func(*args)
  File "/usr/lib/python2.7/urllib2.py", line 556, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
```

**Comment by pricejn2** [comment]
I believe it is `client_id:"2t9loNQH90kzJcsFCODdigxfp325aq4z"`

**Comment by jeremycherfas** [comment]
Does the bookmarklet automatically update to the master code?

**Comment by Hrxn** [comment]
^ 
A new version including the fixed extractor file hasn't been released yet.
You can get the repository via git or download as an archive and run the Python source files with your system's python.

**Comment by Naereen** [comment]
@dstftw Thanks for your quick reply!

**Comment by jeremycherfas** [comment]
Apologies if I was jumping the gun.

**Comment by Hrxn** [comment]
@jeremycherfas (and others)..
dstftw made a new release, please try it again with youtube-dl 2017.03.07 
🥂

**Comment by jeremycherfas** [comment]
Great. Many thanks. Worked perfectly this time.





On Mon, 6 Mar 2017, at 10:57 PM, H R X N wrote:

> @jeremycherfas[1] (and others)..

>  dstftw made a new release, please try it again with youtube-dl
>  2017.03.07
>  🥂



> — You are receiving this because you were mentioned. Reply to this
> email directly, view it on GitHub[2], or mute the thread[3].
> 


Links:

  1. https://github.com/jeremycherfas
  2. https://github.com/rg3/youtube-dl/issues/12376#issuecomment-284546670

**Comment by Walkman100** [comment]
Has it changed again? Couldn't get it to work, and I've just updated it (2017.08.06):
```log
> youtube-dl --verbose https://soundcloud.com/sssshawnnnn/shawn-wasabi-ydg-burnt-rice-henrik-the-artist-remix-the
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'--verbose', u'https://soundcloud.com/sssshawnnnn/shawn-wasabi-ydg-burnt-rice-henrik-the-artist-remix-the']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2017.08.06
[debug] Python version 2.7.6 - Linux-3.19.0-32-generic-x86_64-with-LinuxMint-17.3-rosa
[debug] exe versions: avconv 9.20-6, avprobe 9.20-6, rtmpdump 2.4
[debug] Proxy map: {}
[soundcloud] sssshawnnnn/shawn-wasabi-ydg-burnt-rice-henrik-the-artist-remix-the: Resolving id
[soundcloud] sssshawnnnn/shawn-wasabi-ydg-burnt-rice-henrik-the-artist-remix-the: Downloading info JSON
ERROR: Unable to download JSON metadata: <urlopen error [Errno 1] _ssl.c:510: error:14077410:SSL routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure> (caused by URLError(SSLError(1, '_ssl.c:510: error:14077410:SSL routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure'),))
  File "/usr/lib/python2.7/dist-packages/youtube_dl/extractor/common.py", line 502, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "/usr/lib/python2.7/dist-packages/youtube_dl/YoutubeDL.py", line 2166, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/usr/lib/python2.7/urllib2.py", line 404, in open
    response = self._open(req, data)
  File "/usr/lib/python2.7/urllib2.py", line 422, in _open
    '_open', req)
  File "/usr/lib/python2.7/urllib2.py", line 382, in _call_chain
    result = func(*args)
  File "/usr/lib/python2.7/dist-packages/youtube_dl/utils.py", line 1086, in https_open
    req, **kwargs)
  File "/usr/lib/python2.7/urllib2.py", line 1184, in do_open
    raise URLError(err)
```
I don't think it's a problem with soundcloud blocking me in any way, since I can download it manually, by using the Network tab of the Developer tools to get the stream, then:
```log
> curl "https://api.soundcloud.com/i1/tracks/257889905/streams?client_id=JlZIsxg2hY5WnBgtn3jfS0UYCl0K8DOg"
{"http_mp3_128_url":"https://cf-media.sndcdn.com/wGxOWG9g090T.128.mp3?Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiKjovL2NmLW1lZGlhLnNuZGNkbi5jb20vd0d4T1dHOWcwOTBULjEyOC5tcDMiLCJDb25kaXRpb24iOnsiRGF0ZUxlc3NUaGFuIjp7IkFXUzpFcG9jaFRpbWUiOjE1MDIzMjk3NzB9fX1dfQ__\u0026Signature=L2eSYrOCW-aqyFVPCP5hif3pe3~kz9OBAJ2nDA6uCz49XLY-F3tcSHfwrRUq~OG9hjorJHaSWMHxVlGMi7FZXadFrQ3wwyN4rD5Rp-kWi58eQBw~8pDMwNggQQejCPiTJBTmlCmGqwLlG~y-fSH1masqOwyCf95Y71m~jNsGqeirEDz8o0I-uYEvRXL4zK~YzKzlJGnsscsAV~cJ3muvuTNrxnd~~rd72lgbUaf~ONJh~UDuT3N4Mm3nOtm5c05GUBEpL6pWzbwwa9Lajz19hVvTbFST~SqCNQDGaYZ8FmHT5kvKTx~1~F9iBTDukayLPyM3tkaLcn0cF4zr5eEDXA__\u0026Key-Pair-Id=APKAJAGZ7VMH2PFPW6UQ","hls_mp3_128_url":"https://cf-hls-media.sndcdn.com/playlist/wGxOWG9g090T.128.mp3/playlist.m3u8?Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiKjovL2NmLWhscy1tZWRpYS5zbmRjZG4uY29tL3BsYXlsaXN0L3dHeE9XRzlnMDkwVC4qLiovcGxheWxpc3QubTN1OCIsIkNvbmRpdGlvbiI6eyJEYXRlTGVzc1RoYW4iOnsiQVdTOkVwb2NoVGltZSI6MTUwMjMyOTc3MH19fV19\u0026Signature=dlqhq1gbz8EtoASOQLSpDZVty2MH0r8a6a297Bzau1zovRf1a4XSsBomIYM8rgR551AEIRSdSDdiXXEsZ1215vSZMQrxXJ3u6CSUyfaRSRhsqTryZrhm~lJE7f6AvqJeNAi~T1vcchKpJXIeCTdqikw3bqU7~516YhIe6KE5yY7c5-riO5ALgGkjv-q06GB42J-zjXPseLyjFhCNc4YJMVUZskqMtmYIC0afeq7XahQOt8RAwLoE~0JWsSS9HFuuj69HnjYH2JREE5sXlr30SA2YA987e79QxpZfGFpT1eHPcEmcMN4vMftuUMsbid~EbINg7B4TOHEcgFyneLMYKw__\u0026Key-Pair-Id=APKAJAGZ7VMH2PFPW6UQ","preview_mp3_128_url":"https://cf-preview-media.sndcdn.com/preview/0/30/wGxOWG9g090T.128.mp3?Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiKjovL2NmLXByZXZpZXctbWVkaWEuc25kY2RuLmNvbS9wcmV2aWV3LzAvMzAvd0d4T1dHOWcwOTBULjEyOC5tcDMiLCJDb25kaXRpb24iOnsiRGF0ZUxlc3NUaGFuIjp7IkFXUzpFcG9jaFRpbWUiOjE1MDIzMjk2MDN9fX1dfQ__\u0026Signature=g-vsFRDPfSoxTEB4yHFdg1g5gisV987QhAQVdJ5ySEkJausnFGcUfAe6ncZZ5dlhRRZgso6fIwX7yE6hHAJFw6fP5g8BL2Vpkx5D93WFOReRATvRiwEH090RtSuasN1K-Oy4Mv2OX-RkiltTydvkQVhHUTQBWg7BlQimbAW9szq1lXW7vBtK0McQsAdE013qDB0-KLwB0IVZsvSjA40W7DO~KCE4TCkkhd~ZHphyaOG2WZea-gokdUvmNuhds8K-tqh90alIphIGfR-8F0xCISKtuW2r2cf1V5CgqpV7UicRFfs4-igNiGB~VPnEEsjqhfDQFDxXWnV5BafUoangAQ__\u0026Key-Pair-Id=APKAJAGZ7VMH2PFPW6UQ"}
```
take the first URL, replace the two `\u0026`s with `&`s, then:
```log
> wget "https://cf-media.sndcdn.com/wGxOWG9g090T.128.mp3?Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiKjovL2NmLW1lZGlhLnNuZGNkbi5jb20vd0d4T1dHOWcwOTBULjEyOC5tcDMiLCJDb25kaXRpb24iOnsiRGF0ZUxlc3NUaGFuIjp7IkFXUzpFcG9jaFRpbWUiOjE1MDIzMjk3NzB9fX1dfQ__&Signature=L2eSYrOCW-aqyFVPCP5hif3pe3~kz9OBAJ2nDA6uCz49XLY-F3tcSHfwrRUq~OG9hjorJHaSWMHxVlGMi7FZXadFrQ3wwyN4rD5Rp-kWi58eQBw~8pDMwNggQQejCPiTJBTmlCmGqwLlG~y-fSH1masqOwyCf95Y71m~jNsGqeirEDz8o0I-uYEvRXL4zK~YzKzlJGnsscsAV~cJ3muvuTNrxnd~~rd72lgbUaf~ONJh~UDuT3N4Mm3nOtm5c05GUBEpL6pWzbwwa9Lajz19hVvTbFST~SqCNQDGaYZ8FmHT5kvKTx~1~F9iBTDukayLPyM3tkaLcn0cF4zr5eEDXA__&Key-Pair-Id=APKAJAGZ7VMH2PFPW6UQ" -O shawn-wasabi-ydg-burnt-rice-henrik-the-artist-remix.mp3
--2017-08-10 03:45:42--  https://cf-media.sndcdn.com/wGxOWG9g090T.128.mp3?Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiKjovL2NmLW1lZGlhLnNuZGNkbi5jb20vd0d4T1dHOWcwOTBULjEyOC5tcDMiLCJDb25kaXRpb24iOnsiRGF0ZUxlc3NUaGFuIjp7IkFXUzpFcG9jaFRpbWUiOjE1MDIzMjk3NzB9fX1dfQ__&Signature=L2eSYrOCW-aqyFVPCP5hif3pe3~kz9OBAJ2nDA6uCz49XLY-F3tcSHfwrRUq~OG9hjorJHaSWMHxVlGMi7FZXadFrQ3wwyN4rD5Rp-kWi58eQBw~8pDMwNggQQejCPiTJBTmlCmGqwLlG~y-fSH1masqOwyCf95Y71m~jNsGqeirEDz8o0I-uYEvRXL4zK~YzKzlJGnsscsAV~cJ3muvuTNrxnd~~rd72lgbUaf~ONJh~UDuT3N4Mm3nOtm5c05GUBEpL6pWzbwwa9Lajz19hVvTbFST~SqCNQDGaYZ8FmHT5kvKTx~1~F9iBTDukayLPyM3tkaLcn0cF4zr5eEDXA__&Key-Pair-Id=APKAJAGZ7VMH2PFPW6UQ
Resolving cf-media.sndcdn.com (cf-media.sndcdn.com)... 54.230.199.11
Connecting to cf-media.sndcdn.com (cf-media.sndcdn.com)|54.230.199.11|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 3145977 (3,0M) [audio/mpeg]
Saving to: ‘shawn-wasabi-ydg-burnt-rice-henrik-the-artist-remix.mp3’

100%[=====================================================================================================================================================================================================================================>] 3 145 977    718KB/s   in 4,5s

2017-08-10 03:45:48 (678 KB/s) - ‘shawn-wasabi-ydg-burnt-rice-henrik-the-artist-remix.mp3’ saved [3145977/3145977]
```

**Comment by Hrxn** [comment]
Works for me right now..

Don't know, looks like either issue with certs or the rather old Python version...

**Comment by Walkman100** [comment]
Also, sorry for necroing this thread, but I didn't want to create a new one if it was a simple error on my side or the same issue reoccuring...
@Hrxn what python version does yours report? I have Python3, and my python2.7 is apparently up to date:
```log
> apt install python2.7
Reading package lists... Done
Building dependency tree
Reading state information... Done
python2.7 is already the newest version.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
> youtube-dl --version
2017.08.06
> python2 --version
Python 2.7.6
> python3 --version
Python 3.4.3
> which youtube-dl
/usr/bin/youtube-dl
> python3 /usr/bin/youtube-dl --version
Traceback (most recent call last):
  File "/usr/bin/youtube-dl", line 3, in <module>
    import youtube_dl
ImportError: No module named 'youtube_dl'
```

**Comment by Hrxn** [comment]
My test was done with Python 3.6.1 on Windows 10.

Not sure, but on Windows there has been this issue where youtube-dl/Python would error out because of outdated certs in the cert store.

And the latest Python 2 version is 2.7.13, by the way.

I think 2.7.6 is rather old.

**Comment by Walkman100** [comment]
So I followed [these instructions](https://unix.stackexchange.com/a/110163/102079) to install latest python2.7 to `/opt/python` and now:
```log
> /opt/python/bin/python --version
Python 2.7.13
> /opt/python/bin/python /usr/bin/youtube-dl --version
Traceback (most recent call last):
  File "/usr/bin/youtube-dl", line 3, in <module>
    import youtube_dl
ImportError: No module named youtube_dl
```
I don't know enough about python to find the youtube_dl module, I don't know where it's stored. the contents of `/usr/bin/youtube-dl` is just:
```
#!/usr/bin/python

import youtube_dl

if __name__ == '__main__':
    youtube_dl.main()
```
I don't know how to point it in the right direction, or even where that is. I tried changing the shebang in the first line to `#!/opt/python/bin/python` but I get exactly the same error. Also it's the same error when trying to run with python3 above

**Comment by Walkman100** [comment]
Ok I did `sudo curl -L https://yt-dl.org/downloads/latest/youtube-dl -o /opt/python/bin/youtube-dl`:
```
> export PATH=/opt/python/bin
> python --version
Python 2.7.13
> youtube-dl --version
2017.08.09
> youtube-dl --verbose https://soundcloud.com/sssshawnnnn/shawn-wasabi-ydg-burnt-rice-henrik-th
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'--verbose', u'https://soundcloud.com/sssshawnnnn/shawn-wasabi-ydg-burnt-rice-henrik-th']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2017.08.09
[debug] Python version 2.7.13 - Linux-3.19.0-32-generic-x86_64-with-debian-jessie-sid
[debug] exe versions: none
[debug] Proxy map: {}
[soundcloud] sssshawnnnn/shawn-wasabi-ydg-burnt-rice-henrik-th: Resolving id
[soundcloud] sssshawnnnn/shawn-wasabi-ydg-burnt-rice-henrik-th: Downloading info JSON
ERROR: Unable to download JSON metadata: HTTP Error 404: Not Found (caused by HTTPError()); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
  File "/opt/python/bin/youtube-dl/youtube_dl/extractor/common.py", line 502, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "/opt/python/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 2166, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/opt/python/lib/python2.7/urllib2.py", line 435, in open
    response = meth(req, response)
  File "/opt/python/lib/python2.7/urllib2.py", line 548, in http_response
    'http', request, response, code, msg, hdrs)
  File "/opt/python/lib/python2.7/urllib2.py", line 473, in error
    return self._call_chain(*args)
  File "/opt/python/lib/python2.7/urllib2.py", line 407, in _call_chain
    result = func(*args)
  File "/opt/python/lib/python2.7/urllib2.py", line 556, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
```
TLDR: it's not the python version.

**Comment by dstftw** [comment]
Wrong. https://soundcloud.com/sssshawnnnn/shawn-wasabi-ydg-burnt-rice-henrik-th is 404 that is clearly reported in log.
Your problem is outdated python.

**Comment by Walkman100** [comment]
Oh wow, can't believe I missed that. I really don't know how that got mangled. the downloaded file works with python3 and my downloaded python, I'll make it work, thanks!
